pkg/cmd/experimental/login/smart_merge.go

```diff
@@ -0,0 +1,142 @@
+package login
```

Comment:
If moving upstream it would be under `pkg/client/clientcmd`, correct? Just make sure all important methods are public so we can make use of it.